stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -329,7 +329,7 @@ function process_overrides(artifact_dict::Dict, pkg_uuid::Base.UUID)
 
         for name in keys(artifact_dict)
             # Skip names that we're not overriding
-            if !haskey(pkg_overrides, name)
+            if !(haskey(pkg_overrides, name)::Bool)
```

Comment:
Is `artifact_dict` and `pkg_overrides` always Dict{String, <: Any}` ? Maybe assert that instead?

Reply:
In the case I followed with `ascend`, `artifact_dict` was a `Dict{String, <: Any}`. I am not sure about `pkg_overrides` since I do not know this code in detail - is it?

Reply:
Looks like it is `Dict{String,Union{String,SHA1}}}` according to https://github.com/JuliaLang/julia/blob/99225ab1d11925b8e904b0b12f1ee8b0f6946660/stdlib/Artifacts/src/Artifacts.jl#L94

Reply:
I'm building Julia locally with your suggestion to check whether it fixes the invalidations, too.

Reply:
Okay, looks good - I applied your sugggestion in add3e15bb5250caad664ec1be88aa02d656ab8b6

Reply:
@KristofferC Does the updated version look good to you?